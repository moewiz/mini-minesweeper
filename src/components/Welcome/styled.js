import styled from "styled-components";
import { Link } from "react-router-dom";

export const HomeWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
  margin-top: 30px;
`;

export const GroupLevelStyled = styled.div`
  display: flex;
  flex-direction: row;
  width: 350px;
  justify-content: space-around;
`;

export const LevelBtnStyled = styled(Link)`
  font-size: 20px;
  color: #000;
  border: 1px solid;
  padding: 15px;
  text-decoration: none;
  display: flex;
  &:hover {
    text-decoration: underline;
  }
`;
