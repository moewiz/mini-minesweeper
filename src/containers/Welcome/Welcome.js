import React from "react";
import { Welcome } from "../../components";

const WelcomeContainer = () => {
  return <Welcome />;
};

export default WelcomeContainer;
