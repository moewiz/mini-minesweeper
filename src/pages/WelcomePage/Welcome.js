export { Welcome as default } from "../../containers";
