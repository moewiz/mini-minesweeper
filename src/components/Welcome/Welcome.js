import React, { useEffect } from "react";
import PropTypes from "prop-types";
import { HomeWrapper, GroupLevelStyled, LevelBtnStyled } from "./styled";

const Welcome = ({ newGame }) => {
  useEffect(() => {
    newGame();
  }, [newGame]);

  return (
    <HomeWrapper>
      <h3>Please select game level</h3>
      <GroupLevelStyled>
        <LevelBtnStyled to="/beginner">Beginner</LevelBtnStyled>
        <LevelBtnStyled to="/advantage">Advantage</LevelBtnStyled>
      </GroupLevelStyled>
    </HomeWrapper>
  );
};

Welcome.propTypes = {
  newGame: PropTypes.func.isRequired
};

export default Welcome;
