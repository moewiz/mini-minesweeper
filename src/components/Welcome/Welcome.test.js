import React from "react";
import { mount } from "enzyme";
import { BrowserRouter as Router, Link } from "react-router-dom";
import { act } from "react-dom/test-utils";
import Welcome from "./Welcome";

describe("<Welcome />", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  const newGameMock = jest.fn();
  const Component = mount(
    <Router>
      <Welcome newGame={newGameMock} />
    </Router>
  );

  it("should render without throwing an error", () => {
    expect(Component.length).toBe(1);
  });

  it("should match snapshot", () => {
    expect(Component).toMatchSnapshot();
  });

  it("should call newGameMock 1 time when component did mount", () => {
    // trigger useEffect
    act(() => {
      mount(
        <Router>
          <Welcome newGame={newGameMock} />
        </Router>
      );
    });
    expect(newGameMock).toBeCalledTimes(1);
  });

  it("should render 2 Link", () => {
    const Links = Component.find(Link);
    expect(Links.length).toBe(2);
  });
});
