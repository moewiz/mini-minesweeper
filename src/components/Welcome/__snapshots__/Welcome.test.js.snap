// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Welcome /> should match snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Welcome
      newGame={[MockFunction]}
    >
      <styled.div>
        <div
          className="sc-AxjAm brmHuN"
        >
          <h3>
            Please select game level
          </h3>
          <styled.div>
            <div
              className="sc-AxirZ knUpuA"
            >
              <Styled(Link)
                to="/beginner"
              >
                <Link
                  className="sc-AxiKw bIYgMA"
                  to="/beginner"
                >
                  <LinkAnchor
                    className="sc-AxiKw bIYgMA"
                    href="/beginner"
                    navigate={[Function]}
                  >
                    <a
                      className="sc-AxiKw bIYgMA"
                      href="/beginner"
                      onClick={[Function]}
                    >
                      Beginner
                    </a>
                  </LinkAnchor>
                </Link>
              </Styled(Link)>
              <Styled(Link)
                to="/advantage"
              >
                <Link
                  className="sc-AxiKw bIYgMA"
                  to="/advantage"
                >
                  <LinkAnchor
                    className="sc-AxiKw bIYgMA"
                    href="/advantage"
                    navigate={[Function]}
                  >
                    <a
                      className="sc-AxiKw bIYgMA"
                      href="/advantage"
                      onClick={[Function]}
                    >
                      Advantage
                    </a>
                  </LinkAnchor>
                </Link>
              </Styled(Link)>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </Welcome>
  </Router>
</BrowserRouter>
`;
